geth/rpc/call_raw.go

```diff
@@ -168,13 +168,6 @@ func newSuccessResponse(result json.RawMessage, id json.RawMessage) string {
 		id = defaultMsgID
 	}
 
-	// original response with `null` is lost due to geth RPC client
-	// conversion (unmarshalling). We'll have nil result object in this case,
-	// so handle it specially. `result` is required field in response.
-	if result == nil {
-		result = json.RawMessage("null")
-	}
-
```

Comment:
I think we still need it here. The problem is that we receive `result` object from https://github.com/status-im/status-go/blob/develop/geth/rpc/client.go#L90, which receives it from 
https://github.com/status-im/status-go/blob/develop/vendor/github.com/ethereum/go-ethereum/rpc/client.go#L275, where json marshalling occurs and type `jsonrpcMessage` (defined `vendor/github.com/ethereum/go-ethereum/rpc/client.go`) actually has `omitempty` tag set for `Result` field, so it's just omitted, while expected result should be `null`.

Reply:
Hmm, not, just double checked the test for that case (`go test -v -test.run SuccessResponse/null_nil`), it works fine because nil `json.RawMessage` serializes into `null` anyway. Good spot!